#! /bin/sh -e

# DP: Description: Remove the use of fp regs during the loader
# DP: Author: Randolph Chung  <tausq@debian.org>
# DP: Upstream status: Submitted by Carlos O'Donell <carlos@baldric.uwo.ca>
# DP: Status Details: Awaiting upstream approval
# DP: Date: November 16, 2002

#
#2002-11-11  Randolph Chung  <tausq@debian.org>
#
#        * sysdeps/hppa/Makefile: Disable the use of fp regs in rtld.c
#

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- glibc/sysdeps/hppa/Makefile.orig	2002-09-11 20:19:15.000000000 -0700
+++ glibc/sysdeps/hppa/Makefile	2002-09-11 20:19:17.000000000 -0700
@@ -27,6 +27,7 @@
 endif
 
 ifeq ($(subdir),elf)
+CFLAGS-rtld.c += -mdisable-fpregs
 dl-routines += dl-symaddr dl-fptr
 rtld-routines += dl-symaddr dl-fptr
 endif
